Simplenote/Classes/SPAuthViewController.swift

```diff
@@ -696,131 +799,27 @@ extension SPAuthViewController: SPTextInputViewDelegate {
             state.username = textInput.text ?? ""
         case passwordInputView:
             state.password = textInput.text ?? ""
+        case codeInputView:
+            state.code = textInput.text ?? ""
         default:
             break
         }
     }
 
     func textInputShouldReturn(_ textInput: SPTextInputView) -> Bool {
-        switch textInput {
-        case emailInputView:
-            switch performUsernameValidation() {
-            case .success:
-                if mode.isPasswordHidden {
-                    performPrimaryActionIfPossible()
-                } else {
-                    passwordInputView.becomeFirstResponder()
-                }
-
-            case let error:
-                displayEmailValidationWarning(error.description)
-            }
-
-        case passwordInputView:
-            switch performPasswordValidation() {
-            case .success:
-                performPrimaryActionIfPossible()
-
-            case let error:
-                displayPasswordValidationWarning(error.description)
-            }
-
-        default:
-            break
-        }
-
+        performPrimaryActionIfPossible()
         return false
     }
 }
 
 
-// MARK: - State
```

Comment:
All of the structs below have been extracted + simplified into `AuthenticationMode`